client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/HeaderDecoder.java

```diff
@@ -369,8 +369,12 @@ private void readNewTopologyAndHash(ByteBuf buf, String cacheName) {
          segmentOwners = null;
       }
 
-      dispatcher.updateTopology(cacheName, operation, newTopologyId,
-            addresses, segmentOwners, hashFunctionVersion);
+      // Only update the topology if this channel is fully connected and authenticated
```

Comment:
How often server sends topology updates ?

Reply:
The topology is sent back as a response every time the client sends a request that is not the same one.

Reply:
What you are talking about is if all of the IPs are no longer accessible in the topology provided. In that case the client resets its server list back to the initial hosts list and effectively "resets". This transition should work seamlessly over as it has in the past.

Reply:
@wburns , 
Client ignores topology updates if the server perspective of topology has not changed -  
```
case READ_TOPOLOGY:
               short topologyChangeByte = in.readUnsignedByte();
               if (topologyChangeByte == 1) {
                  readNewTopologyAndHash(in, operation.getCacheName());
               }
```

Does that mean that by ignoring topology updates when the channel is not accepting requests, the channel might remain stuck with an outdated topology?

Reply:
We only ignore topology updates while the channel is initially connecting. This only happens very briefly at the start, in the case of when auth is enabled it is until the auth is complete (without auth it is just the initial ping). Any response after that will apply the new topology. Since the request has an older topology _every_ response will contain the new topology, until it is finally applied.

Reply:
What do you think about comparing response topologyId with `cacheInfo.getTopologyId()`  instead of relying on topologyChangeByte ?

Reply:
The server already does this comparison and if they don't match includes it in the response. This way the server only includes this information in the response when they don't match.

https://github.com/infinispan/infinispan/blob/main/server/hotrod/src/main/java/org/infinispan/server/hotrod/Encoder2x.java#L696